### ISSUE TITLE
Need to add custom unit tests to ensure the current code does not break

### ISSUE THREAD

**Issue by bryantson** [mona-actions/gh-repo-stats#25]
If more changes are introduced to the current script, stable script can break. Although an extensive Pull Request review can prevent that, it is hard to ensure what works or not without an extensive guideline and testing.

There should be an additional CI pipeline and associated scripts to automate the code testing process.

**Comment by ssulei7** [comment]
@bryantson agreed. I'd like to see some integration tests built in bash where we can validate output, etc.

**Comment by JSON2MAZA** [comment]
Ya ya

**Comment by hafifudin** [comment]
1. 

> 

- [ ] [****](url)

**Comment by hafifudin** [comment]
- ### 

> [`_****_`](url)

**Comment by drellwyatt5-svg** [comment]
> If more changes are introduced to the current script, stable script can break. Although an extensive Pull Request review can prevent that, it is hard to ensure what works or not without an extensive guideline and testing.
> 
> There should be an additional CI pipeline and associated scripts to automate the code testing process.
